zml/aio/torch.zig

```diff
@@ -314,79 +314,76 @@ pub const PickleData = struct {
+                                else => unreachable,
+                            };
+                            switch (valid_slice) {
+                                true => try store._metadata.put(
+                                    allocator,
+                                    try allocator.dupe(u8, prefix.items),
+                                    .{ .array = .{ .item_type = used_tag, .data = std.mem.sliceAsBytes(try values.toOwnedSlice(allocator)) } },
+                                ),
+                                false => for (values.items, 0..) |val, i| {
+                                    var new_prefix = prefix;
+                                    if (prefix.items.len > 0) {
+                                        new_prefix.appendAssumeCapacity('.');
+                                    }
+                                    new_prefix.items.len += std.fmt.formatIntBuf(new_prefix.unusedCapacitySlice(), i, 10, .lower, .{});
+                                    try store._metadata.put(allocator, try allocator.dupe(u8, new_prefix.items), @unionInit(zml.aio.Value, @tagName(used_tag), val));
+                                },
+                            }
+                        },
+                        else => {
+                            for (seq[1], 0..) |item, i| {
+                                var new_prefix = prefix;
+                                if (v.isPrimitive()) {
+                                    if (prefix.items.len > 0) {
+                                        new_prefix.appendAssumeCapacity('.');
+                                    }
+                                    new_prefix.items.len += std.fmt.formatIntBuf(new_prefix.unusedCapacitySlice(), i, 10, .lower, .{});
                                 }
-                                new_prefix.items.len += std.fmt.formatIntBuf(new_prefix.unusedCapacitySlice(), i, 10, .lower, .{});
+                                try self.parseValue(allocator, store, new_prefix, item);
                             }
-                            try self.parseValue(allocator, store, new_prefix, item);
-                        }
+                        },
                     }
                 },
-                .dict => {
-                    for (seq[1]) |item| {
-                        try self.parseValue(allocator, store, prefix, item);
-                    }
+                .dict => for (seq[1]) |item| {
+                    try self.parseValue(allocator, store, prefix, item);
                 },
                 .kv_tuple => {
                     const key = seq[1][0];
                     const val = seq[1][1];
                     switch (key) {
                         .string => |s| {
-                            if (std.mem.eql(u8, s, "_modules") or std.mem.eql(u8, s, "_parameters") or std.mem.eql(u8, s, "_buffers")) {
+                            if (std.meta.stringToEnum(enum { _modules, _parameters, _buffers }, s)) |_| {
```

Comment:
not fond of this stringToEnum hack. the std.mem.eql are fine. You can make a small util at the bottom of the file if you want to shorten it.
also you should document those are pytorch specific fields.

Reply:
Fair enough; will revert to `std.mem.eql`